### PR TITLE
Feature/gsim logical record relation

### PR DESCRIPTION
**Pull request by RuneLind**
Implement parent child relationship for GSIM LogicalRecors. 
Tested in localstack

### CODE REVIEW

File: src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java [hunk 1]
```diff
@@ -56,20 +54,22 @@ public void createGsimObjects() {
...
         LogicalRecord gsimLogicalRecord =
-                createDefault(createId(logicalRecord), logicalRecord.getName(), null)
+                createDefault(logicalRecordId, logicalRecord.getName(), null)
                         .logicalRecord()
                         .isPlaceholderRecord(false)// TODO: add and get from simple
                         .unitType(logicalRecord.getUnitType(), "UnitType_DUMMY")
                         .shortName(logicalRecord.getName())
                         .instanceVariables(logicalRecord.getInstanceVariableIds(i -> createId(logicalRecord, i)))
+                        .parent(parentLogicalRecordId)
+                        .parentChildMultiplicity("ONE_MANY")
```
Comment (bjornandre) [inline src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java:70]: Should this be an enum? I would prefer to use "ONE_TO_MANY"

File: src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java [hunk 2]
```diff
@@ -56,20 +54,22 @@ public void createGsimObjects() {
                 .build();
         persistenceProvider.save(unitDataset);
 
-        processAll(rootLogicalRecord, 0);
+        processAll(rootLogicalRecord, null, 0);
     }
 
-    void processAll(no.ssb.dapla.dataset.doc.model.simple.LogicalRecord logicalRecord, int level) {
+    void processAll(no.ssb.dapla.dataset.doc.model.simple.LogicalRecord logicalRecord, String parentLogicalRecordId, int level) {
```
Comment (bjornandre) [inline src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java:60]: level doesn't seem to be used anymore, right?

Comment (RuneLind) [reply src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java:60]: True, was used for debugging. Will remove 👍

File: src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java (hunk 1 above)
Comment (RuneLind) [reply src/main/java/no/ssb/dapla/dataset/doc/template/SimpleToGsim.java:70]: This is from the enum in the ramel definition of LogicalRecord: https://github.com/statisticsnorway/gsim-raml-schema/blob/4e27cdb7a533e40c01e3d6fd3d57a764a10412c4/schemas/LogicalRecord.raml
We could make a enum mirroring this as well for better readability

**Pull request merged**